File: web_login_flutter/lib/models/form/text_question.dart
import 'package:web_login_flutter/models/form/question.dart';

class TextQuestion extends Question {
  final Question? _nextQuestion;
  const TextQuestion({required super.id, required super.title, required super.description, required super.imageUrl, required super.isMandatory, nextQuestion}):
    _nextQuestion = nextQuestion;

  @override
  String toString() {
    return '${super.toString()}TextQuestion{_nextQuestion: $_nextQuestion}';
  }

  Question? get nextQuestion => _nextQuestion;

  bool get hasNextQuestion => _nextQuestion != null;

  @override
  List<Object?> get props => [];
}

File: web-login-core/src/main/java/com/example/weblogincore/domain/model/form/questions/TextQuestion.java
package com.example.weblogincore.domain.model.form.questions;

public class TextQuestion extends Question {

    private final Question nextQuestion;

    public TextQuestion(Long id, String title, String description, String imageUrl, Boolean isMandatory, Question nextQuestion) {
        super(id, title, description, imageUrl, isMandatory);
        this.nextQuestion = nextQuestion;
    }

    public Question getNextQuestion() {
        return nextQuestion;
    }

    @Override
    public Integer totalQuestions() {
        if (nextQuestion == null ) return 1;

        return 1 + nextQuestion.totalQuestions();
    }
}
